spec/hardware/system.fmf

```diff
@@ -20,6 +20,13 @@ description: |
            # Integer or string, required number of NUMA nodes.
            numa-nodes: 2|">= 2"
 
+           # String, which labcontroller to use or to avoid.
+           # Note: this field is consumed by Beaker and Artemis plugins only.
+           lab-controller: "lab-01.foo.bar.com"
```

Comment:
Hi @happz , is there examples how does Beaker or Artmis plugin use lab-controller? If my understanding is correct, this is a network related field instead of user defined field ?

Reply:
Both `beaker` and `artemis` provision plugins would take the value and pass it into Beaker XML, eventually you might spot something like this in the job XML:

```xml
<labcontroller op="=" value="lab-01.foo.bar.com"/>
<labcontroller op="!=" value="lab-01.foo.bar.com"/>
<labcontroller op="like" value="lab-%.foo.bar.com"/>
```

I wouldn't say it's a network-related field, at least not related to the network of the provisioned guest itself. Yes, it may result in guest having some properties that wouldn't be available with a different lab controller (e.g. IPv6 connectivity), but there may be other non-network effects, e.g. specific HW, models, availability, etc. Also, I wouldn't say it's a user-defined field, it's a feature offered by Beaker, similar to "availability zone" and "region" in AWS and other cloud providers, and the values must make some sense to Beaker service accessed by the `beaker` plugin. We may decide how the key should look like and where it should be placed, but down the stream, it'd be `<labcontroller/>` element.

Reply:
Resolving